ios/FluentUI/Core/UIKit+SwiftUI_interoperability.swift

```diff
@@ -16,18 +16,14 @@ struct UIViewAdapter: UIViewRepresentable {
     }
 
     func makeUIView(context: Context) -> UIView {
-        return makeView()
-    }
+        let view = makeView()
 
-    func updateUIView(_ view: UIView, context: Context) {
-        // This logic should be removed wnce the "wrapping in a UIStackView" workaround is removed.
-        guard let stackView = view as? UIStackView else {
-            return
-        }
+        // Ensures that the UIView respects the frame defined in the SwiftUI layer.
+        view.setContentCompressionResistancePriority(.defaultLow, for: .vertical)
+        view.setContentCompressionResistancePriority(.defaultLow, for: .horizontal)
 
-        for view in stackView.arrangedSubviews {
```

Comment:
This was introduced in #442, I was trying to brainstorm another solution for this without using the StackView. Will we work this out in a separate PR?

Reply:
Thanks a lot for bringing this up @sophialee0416!
I have experimented some options and concluded we do need a "container" view.
The view we return in makeUIView isis never replaced/updated even when we do `UIViewAdapter(myUIview)` in the SwiftUI enviroment. So, we just replace the contents of that container view.
A UIStackView provides us the constraint and compression resistance functionality for free, which is why I think using it is probably the best option we have now.